File: 347-top-k-frequent-elements/top-k-frequent-elements.py
class Solution(object):
    def topKFrequent(self, nums, k):
        counts = {}

        for num in nums:
            if num in counts:
                counts[num] +=1
            else:
                counts[num] = 1

        sortedCounts = sorted(counts.items(),key=lambda item: item[1], reverse=True)

        return [item[0] for item in sortedCounts[:k]]

        

File: 347-top-k-frequent-elements/top-k-frequent-elements.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int,int> count;

        for(int n : nums){
            count[n]++;
        }
        priority_queue<pair<int, int>> pq;
        for (auto it : count) {
            pq.push({it.second, it.first}); 
        }
        vector<int> result;
        for (int i = 0; i < k; i++) {
            result.push_back(pq.top().second);  
            pq.pop();
        }
        return result;

    }
};